website/static/js/pages/file-page.js

```diff
@@ -4,4 +4,44 @@ var $osf = require('js/osfHelpers');
 var FileViewPage = require('js/filepage');
 var waterbutler = require('js/waterbutler');
 
+require('jquery-tagsinput');
+
 m.mount(document.getElementsByClassName('file-view-panels')[0], FileViewPage(window.contextVars));
+
+function build_tag_url(nid, filePath, tag) {
+    return '/api/v1/project/' + nid + '/osfstorage' + filePath + '/tags/' + tag + '/';
+}
+
+$(function() {
+    // Tag input
+    $('#fileTags').tagsInput({
+        width: '100%',
+        interactive: window.contextVars.currentUser.canEdit,
+        maxChars: 128,
+        onAddTag: function (tag) {
+            var url = build_tag_url(window.contextVars.node.id, window.contextVars.file.path, tag);
```

Comment:
The url path includes the tag itself, so that would only be able to build part of the url, and then it would be called like ` url = tagUrl + tag + '/'; ` If you feel that is an improvement, I can make that change also.